src/main/java/teammates/ui/webapi/action/LinkRecoveryAction.java

```diff
@@ -0,0 +1,40 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.output.LinkRecoveryResponseData;
+
+
+/**
+ * Action specifically created for confirming email and sending session recovery links.
+ */
+public class LinkRecoveryAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // no specific access control needed.
+    }
+
+    @Override
+    public ActionResult execute() {
+        String recoveryEmail = getNonNullRequestParamValue(Const.ParamsNames.RECOVERY_EMAIL);
+        boolean hasStudentsWithRecoveryEmail = !logic.getAllStudentForEmail(recoveryEmail).isEmpty();
+
+        if (hasStudentsWithRecoveryEmail) {
+            EmailWrapper email = new EmailGenerator().generateLinkRecoveryEmail(recoveryEmail);
+            emailSender.sendEmail(email);
+            return new JsonResult(new LinkRecoveryResponseData(true,
+                    "The recovery links for your feedback sessions have been sent to the specified email: "
+                            + recoveryEmail));
+        } else {
+            return new JsonResult(new LinkRecoveryResponseData(false,
+                    "No student is registered under email: " + recoveryEmail));
```

Comment:
@junming403 Maybe we should remove this message based on what @damithc stated [here](https://github.com/TEAMMATES/teammates/pull/8543#issuecomment-379673691). 
> * We should not tell the user if the email is a valid TEAMMATES user because we don't want to leak that information to an unauthenticated user

You can still keep the boolean value (for status message, if recaptcha verification fails), but we'll rename it later.

Reply:
I will leave the flag there for you to use.